docs/design/2022-12-07-keyspace.md

```diff
@@ -0,0 +1,200 @@
+# Proposal: Keyspace
+* Authors: [ystaticy](https://github.com/ystaticy), [iosmanthus](https://github.com/iosmanthus), [AmoebaProtozoa](https://github.com/AmoebaProtozoa),[zeminzhou](https://github.com/zeminzhou)
+
+
+## Motivation
+
+We aim to store data from multiple applications in a TiKV cluster. For instance, in a multi tenant scenario, a common solution is to add a prefix to the key to differentiate the data of various applications. Therefore, we introduce a new concept referred to as `Keyspace` to describe the logical isolation among different business scenarios within a TiKV cluster.
+
+## Architecture
+
+In this section, we will outline the architecture when utilizing keyspaces.
+
+![keyspace-arch.png](imgs/keyspace-arch.png)
+
+### PD
+
+PD is responsible for managing `Keyspace` metadata. It provides HTTP and RPC interfaces to create, load, and modify `Keyspace` metadata. All other components retrieve the keyspace metadata from PD.
+
+### TiDB
+
+After the keyspace is created on PD, it can be configured on multiple TiDB instances, forming a TiDB cluster for that keyspace. However, a TiDB instance can serve only one keyspace at a specific moment, unless it is restarted and configured to use a different keyspace.
+
+### Client-go
+
+In TiDB, client-go utilizes codec V2 to add a keyspace prefix when the keyspace is set. A client configured with a keyspace can only read and write data from the specified keyspace.
+
+### TiKV
+
+In TiKV, different keyspaces are isolated by different data key prefixes.
+
+### TiFlash
+
+In TiFlash, requests carry the keyspace prefix, and the data structures and storage paths are isolated by the keyspace ID.
+
+## Usage
+
+To use TiDB with `Keyspace`, the following configuration is required:
+1. TiKV needs to [enable API V2](https://github.com/tikv/rfcs/blob/master/text/0069-api-v2.md#tikv-server).
+2. The TiDB configuration should set `keyspace-name` to specify which keyspace is currently served by the TiDB instance.
+3. BR, Lightning, Dumpling, and CDC specify the keyspace by setting `keyspace-name` on the command line.
```

Comment:
```suggestion
3. BR, Lightning, Dumpling, and TiCDC specify the keyspace by setting `keyspace-name` on the command line.
```

Reply:
Is 'keyspace' configuration needed by Dumpling?

Reply:
> Is 'keyspace' configuration needed by Dumpling?

Dumpling also requires the keyspace-name parameter to verify whether the keyspace of the connected TiDB matches the expected one.